abcix/adapter/adapter.go

```diff
@@ -234,6 +266,50 @@ func (app *adaptedApp) ApplySnapshotChunk(req abcix.RequestApplySnapshotChunk) (
 	return
 }
 
-func AdaptToABCIx(abciApp abci.Application) abcix.Application {
-	return &adaptedApp{abciApp: abciApp}
+func AdaptToABCIx(abciApp abci.Application, optionDB ...dbm.DB) abcix.Application {
+	var db dbm.DB
+	if len(optionDB) == 0 {
+		dir, err := ioutil.TempDir("/tmp", "adaptor")
+		if err != nil {
+			panic("failed to generate adaptor DB dir")
+		}
+
+		defer os.RemoveAll(dir)
+		db, err = dbm.NewGoLevelDB("adaptor", dir)
+		if err != nil {
+			panic("failed to generate adaptor DB")
+		}
+	} else {
+		db = optionDB[0]
+	}
+	state := loadState(db)
+	return &adaptedApp{abciApp: abciApp, state: state}
+}
+
+func loadState(db dbm.DB) state {
+	var state state
+	state.db = db
+	stateBytes, err := db.Get(stateKey)
+	if err != nil {
+		panic(err)
```

Comment:
panic during loading makes sense (during start up), but saving state could happen during normal running. I would suggest we handle it more gracefully, e.g. only print error logs